File: app/src/main/java/com/orangepenguin/boilerplate/usecase/UserUseCase.java
package com.orangepenguin.boilerplate.usecase;

import com.orangepenguin.boilerplate.model.User;
import com.orangepenguin.boilerplate.repository.UserRepository;

import javax.inject.Inject;

import rx.Observable;

public class UserUseCase {

    private final UserRepository userRepository;

    @Inject
    public UserUseCase(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public Observable<User> fetchUser(String username) {
        return userRepository.fetchUser(username);
    }
}

File: app/src/main/java/com/orangepenguin/boilerplate/usecase/UserUseCase.kt
package com.orangepenguin.boilerplate.usecase

import com.orangepenguin.boilerplate.model.User
import com.orangepenguin.boilerplate.repository.UserRepository
import rx.Observable
import javax.inject.Inject

open class UserUseCase @Inject constructor(private val userRepository: UserRepository) {

    open fun fetchUser(username: String): Observable<User> {
        return userRepository.fetchUser(username)
    }
}
